### DISCUSSION TITLE
Time cost to get sorted_tasks

### DISCUSSION THREAD

**Discussion by zqbake** [PrefectHQ/prefect#4214] [Q&A] [answered]
https://github.com/PrefectHQ/prefect/blob/849c903404da1c352453c41d67c9508de1959922/src/prefect/core/flow.py#L879

Why not use the [usual way](https://en.wikipedia.org/wiki/Topological_sorting) to get the sorting ?

**Comment by jcrist** [comment, marked as answer]
Are you having a performance issue with this? We do perform a topological sort here (although there's a few opportunities for perf improvements if needed). If performance isn't an issue, I'm reluctant to change anything here.

**Reply by zqbake** [reply]
No, I am just curious.
